lib/rules/prefer-screen-queries.ts

```diff
@@ -113,52 +136,49 @@ export default createTestingLibraryRule<Options, MessageIds>({
           // save the destructured query methods as safe since they are coming
           // from within or render + base/container options
           saveSafeDestructuredQueries(node);
-          return;
-        }
-
-        if (ASTUtils.isIdentifier(node.id)) {
+        } else if (ASTUtils.isIdentifier(node.id)) {
           withinDeclaredVariables.push(node.id.name);
         }
       },
-      'CallExpression > Identifier'(node: TSESTree.Identifier) {
-        if (!helpers.isBuiltInQuery(node)) {
+      CallExpression(node) {
+        const identifierNode = getDeepestIdentifierNode(node);
+
+        if (!identifierNode) {
           return;
         }
 
-        if (
-          !safeDestructuredQueries.some((queryName) => queryName === node.name)
-        ) {
-          reportInvalidUsage(node);
-        }
-      },
-      'MemberExpression > Identifier'(node: TSESTree.Identifier) {
-        function isIdentifierAllowed(name: string) {
-          return ['screen', ...withinDeclaredVariables].includes(name);
+        if (helpers.isRenderUtil(identifierNode)) {
+          detectRenderWrapper(identifierNode);
         }
 
-        if (!helpers.isBuiltInQuery(node)) {
+        if (!helpers.isBuiltInQuery(identifierNode)) {
           return;
         }
 
-        if (
-          ASTUtils.isIdentifier(node) &&
-          isMemberExpression(node.parent) &&
-          isCallExpression(node.parent.object) &&
-          ASTUtils.isIdentifier(node.parent.object.callee) &&
-          node.parent.object.callee.name !== 'within' &&
-          helpers.isRenderUtil(node.parent.object.callee) &&
-          !usesContainerOrBaseElement(node.parent.object)
-        ) {
-          reportInvalidUsage(node);
-          return;
-        }
+        if (isMemberExpression(identifierNode.parent)) {
```

Comment:
Can we do early returns instead of these nested if statements?

Reply:
I'm afraid not since there are different strategies to follow depending on parent type.

Reply:
What do you think of
```ts
        if (!isMemberExpression(identifierNode.parent)) {
          const isSafeDestructuredQuery = safeDestructuredQueries.some(
            (queryName) => queryName === identifierNode.name
          );
          if (isSafeDestructuredQuery) {
            return;
          }
          
          return reportInvalidUsage(identifierNode);
        }

        const memberExpressionNode = identifierNode.parent;
        if (
          isCallExpression(memberExpressionNode.object) &&
          ASTUtils.isIdentifier(memberExpressionNode.object.callee) &&
          memberExpressionNode.object.callee.name !== 'within' &&
          isReportableRender(memberExpressionNode.object.callee) &&
          !usesContainerOrBaseElement(memberExpressionNode.object)
        ) {
          return reportInvalidUsage(identifierNode);
        }

        if (
          ASTUtils.isIdentifier(memberExpressionNode.object) &&
          !isIdentifierAllowed(memberExpressionNode.object.name)
        ) {
          return reportInvalidUsage(identifierNode);
        }
```

Reply:
It's not perfect, but it has less nested if statements, which is already a good thing I think

Reply:
Updated. I think the code seems simple in terms of nested if statements, but it's not so clear which are the different paths regarding the parent type.